src/workerd/server/server.c++

```diff
@@ -1303,10 +1304,79 @@ private:
   capnp::Capability::Client getCapability(uint channel) override {
     KJ_FAIL_REQUIRE("no capability channels");
   }
+  class CacheClientImpl final: public CacheClient {
+  public:
+    CacheClientImpl(Service& parent)
+        : parent(parent) {
+    }
+
+    kj::Own<kj::HttpClient> getDefault(kj::Maybe<kj::String> cfBlobJson,
+                                       kj::Maybe<Tracer::Span&> parentSpan) override {
+
+      return kj::heap<CacheHttpClientImpl>(
+          parent, nullptr, kj::mv(cfBlobJson), kj::mv(parentSpan));
+    }
+
+    kj::Own<kj::HttpClient> getNamespace(
+        kj::StringPtr cacheName, kj::Maybe<kj::String> cfBlobJson,
+        kj::Maybe<Tracer::Span&> parentSpan) override {
+
+      auto encodedName = kj::encodeUriComponent(cacheName);
+
+      return kj::heap<CacheHttpClientImpl>(
+          parent,  kj::mv(encodedName), kj::mv(cfBlobJson), parentSpan);
+    }
+
+  private:
+    Service& parent;
+  };
+
+  class CacheHttpClientImpl final: public kj::HttpClient {
+  public:
+    CacheHttpClientImpl(Service& parent,
+                        kj::Maybe<kj::String> cacheName, kj::Maybe<kj::String> cfBlobJson,
+                        kj::Maybe<Tracer::Span&> parentSpan)
+        : cacheName(kj::mv(cacheName)),
+          cfBlobJson(kj::mv(cfBlobJson)), parentSpan(kj::mv(parentSpan)) {
+      client = asHttpClient(parent.startRequest({kj::mv(cfBlobJson), parentSpan}));
+    }
+
+    Request request(kj::HttpMethod method, kj::StringPtr url,
+                    const kj::HttpHeaders &headers,
+                    kj::Maybe<uint64_t> expectedBodySize = nullptr) override {
+
+      return client->request(method, url, addCacheNameHeader(headers, cacheName),
+                           expectedBodySize);
+    }
+
+  private:
+    kj::Own<kj::HttpClient> client;
+    kj::Maybe<kj::String> cacheName;
+    kj::Maybe<kj::String> cfBlobJson;
+    kj::Maybe<Tracer::Span&> parentSpan;
+
+    kj::HttpHeaders addCacheNameHeader(const kj::HttpHeaders& headers,
+       kj::Maybe<kj::StringPtr> cacheName) {
+      kj::ArrayPtr<const char> spliceParts[3];
+      uint spliceCount = 0;
+      auto headersCopy = headers.cloneShallow();
+      KJ_IF_MAYBE(name, cacheName) {
+        spliceParts[spliceCount++] = "{\"namespace\":\""_kj;
```

Comment:
I know you copied this splice stuff from internal code, but the place you copied from was a lot more complicated, and in this case you could do something much cleaner:

```
kj::String headerValue;
KJ_IF_MAYBE(name, cacheName) {
  headerValue = kj::str("{\"namespace\":\"", *name, "\"}");
} else {
  headerValue = kj::str("{}");
}
```

Better yet, if the only purpose of this header is to encode the namespace name, maybe don't use JSON at all. You could have the header be named `X-Cache-Namespace` and have it just contain the name raw. If no namespace is specified, then leave the header off.